sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +112,56 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, conf: RapidsConf): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val possibleRapidsJarURLs = classloader.getResources(propName).asScala.toSet.toSeq.filterNot {
+      url => url.toString.contains("test")
+    }
+    val revisionMap: Map[String, Seq[URL]] = possibleRapidsJarURLs.map { url =>
+      val versionInfo = scala.io.Source.fromURL(url).getLines().toSeq
+      lazy val revision = versionInfo
+        .filter(_.startsWith("revision="))
+        .map(_.split("=").last)
+        .headOption
+        .getOrElse("UNKNOWN")
+      (revision, url)
+    }.groupBy(_._1).mapValues(_.map(_._2)).toMap
+    lazy val rapidsJarsVersMsg = revisionMap.map {
+      case (revision, urls) => {
+        s"revison: $revision" + urls.map {
+          url => "\n\tjar URL: " + url.toString.split("!").head + "\n\t" + 
+              scala.io.Source.fromURL(url).getLines().toSeq.mkString("\n\t")
+        }.mkString + "\n"
+      }
+    }.mkString
+    // scalastyle:off line.size.limit
+    lazy val msg = s"""Multiple $jarName jars found in the classpath:
+        |$rapidsJarsVersMsg
+        |Please make sure there is only one $jarName jar in the classpath.
+        |If it is impossible to fix the classpath you can suppress the error by setting ${RapidsConf.ALLOW_MULTIPLE_JARS.key} to SAME_REVISION or ALWAYS.
+        """.stripMargin
+    // scalastyle:on line.size.limit
+
+    conf.allowMultipleJars match {
+      case "ALWAYS" =>
+        logWarning(msg)
+      case "SAME_REVISION" =>
+        require(revisionMap.size == 1, msg)
+      case "NEVER" =>
+        require(revisionMap.size == 1 && revisionMap.values.forall(_.size == 1), msg)
+      case _ => 
+        throw new IllegalArgumentException(s"Invalid value for " +
+          s"${RapidsConf.ALLOW_MULTIPLE_JARS.key}: ${conf.allowMultipleJars}. " +
+          s"Valid values are ALWAYS, SAME_REVISION, NEVER.")
+    }
```

Comment:
let us follow the pattern of other configs with an enum. Take a look PARQUET_READER_FOOTER_TYPE and make an auxiliary function translating strings <-> enum.

Reply:
done